src/server/FileUploadDataSource.ts

```diff
@@ -0,0 +1,191 @@
+import {
```

Comment:
@Herraj @kkyeboah can you add a comment about why we pulled this in?

Reply:
Yes sorry I will add comments, wanted to test it out first.

But the motivation was to extend that class from that library and implement that one line fix from the PR that hasn't been merged to that library's main branch.

In order to do that, I had to pretty much copy the whole thing for it to compile

Reply:
ooff that sucks. thank you for that